### ISSUE TITLE
F523 fix changes behavior by removing `str.format` call

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#15557]
When the fix for [`string-dot-format-extra-positional-arguments` (F523)](https://docs.astral.sh/ruff/rules/string-dot-format-extra-positional-arguments/) removes a string’s `format` method call, it can change the program’s behavior. If the string contains a double brace like `{{`, it originally represents an escape sequence for a single brace like `{`, but with the method call removed it literally represents a double brace. If the string contains an unset field name, removing the method call suppresses the expected `KeyError`.

```console
$ ruff --version
ruff 0.9.2

$ cat f523.py
print("{{".format("!"))
print("{x}".format("!"))

$ python f523.py
{
Traceback (most recent call last):
  File "f523.py", line 2, in <module>
    print("{x}".format("!"))
          ^^^^^^^^^^^^^^^^^
KeyError: 'x'

$ ruff --isolated check --fix --select F523 f523.py
Found 2 errors (2 fixed, 0 remaining).

$ python f523.py
{{
{x}
```

**Comment by knavdeep152002** [comment]
Hey @dscorbett ,

So, what should be the ideal fix for the example you mentioned? 

Should it be

```python
print("{")
print("{x}".format(x="!"))
```

Or is it expected to be something else?

**Comment by dscorbett** [comment]
The simplest fix would be to revert #15309:
```python
print("{{".format())
print("{x}".format())
```

The ideal fix would be:
```python
print("{")
print("{x}".format())
```

For the first expression, the `format` call can be removed if the double braces are unescaped into a single brace. For the second expression, the unused argument can be removed, but the `format` call can’t because that would suppress the `KeyError`.
